operator/v1/types_ingress.go

```diff
@@ -1306,6 +1306,23 @@ type IngressControllerTuningOptions struct {
 	// +kubebuilder:validation:Format=duration
 	// +optional
 	TLSInspectDelay *metav1.Duration `json:"tlsInspectDelay,omitempty"`
+
+	// healthCheckInterval defines how long the router waits between two consecutive
+	// health checks on its configured backends.  This value is applied globally as
+	// a default for all routes, but may be overridden per-route by the route annotation
+	// "router.openshift.io/haproxy.health.check.interval".
+	//
+	// Setting this to less than 5s can cause excess traffic due to too frequent
+	// TCP health checks and accompanying SYN packet storms.  Alternatively, setting
+	// this too high can result in increased latency, due to backend servers that are no
+	// longer available, but haven't yet been detected as such.
+	//
+	// If unset, the default healthCheckInterval is 5s.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Format=duration
+	// +optional
+	HealthCheckInterval *metav1.Duration `json:"healthCheckInterval,omitempty"`
```

Comment:
The `kubebuilder:validation:Format=durationcis a duration string, a "sequence of decimal numbers, each with optional fraction and a unit suffix, such as "300ms", "-1.5h" or "2h45m". Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h"."
Afaik we don't support kubebuilder validation for max/min strings.

Reply:
> Afaik we don't support kubebuilder validation for max/min strings.

but your operator should prevent pathological values  in order to protect the user. What are the limits your operator will enforce?  For instance, 1millisecond is probably too small and 60 minutes is probably too long.  But will you enforce tighter than that?

Reply:
That is something we could take up in the operator if needed.

Reply:
> That is something we could take up in the operator if needed.

The operator is an acceptable place to do this, but it's an API contract that the API client needs to know about.  It should be doccumented here (with specific values) and honored in either the operator or an admission plugin.

Reply:
So, if I add document a max/min value, can it also be marked as "subject to change over time", as was requested for future proofing the default value?

Reply:
@deads2k I made the update here, ptal.

Reply:
The intention of allowing large values (even greater than 60 minutes) is to allow the user effectively to turn off health checks entirely.  0 gets you the default value (= 5s), but we could restrict values between 0 and 1s to avoid making it _too_ easy to break a cluster by setting a value such as 1 millisecond.  

@candita, how about if we say that the allowed range is 1s to 2147483647ms (= 24.85 days), and that 0 is interpreted as the default value (= 5s)?

Reply:
By the way, the 2147483647ms (= 24.85 days) upper bound derives from an HAProxy limitation.  The fact that HAProxy imposes this limit is the only reason we didn't choose a higher or rounder number.

Reply:
1s to 24.85 days works for me.